new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleTraefikValidation.java

```diff
@@ -0,0 +1,685 @@
+            domainUID, domainNamespace));
+
+    // wait for the domain to exist
+    logger.info("Check for domain custom resouce in namespace {0}", domainNamespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for domain {0} to be created in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                domainUID,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainExists(domainUID, DOMAIN_VERSION, domainNamespace));
+
+    // check admin server pod exist
+    logger.info("Check for admin server pod {0} existence in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodCreated(adminServerPodName);
+
+    // check managed server pods exists
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Check for managed server pod {0} existence in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkPodCreated(managedServerPrefix + i);
+    }
+
+    // check admin server pod is running
+    logger.info("Wait for admin server pod {0} to be ready in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReady(adminServerPodName);
+
+    // check managed server pods are running
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server pod {0} to be ready in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkPodReady(managedServerPrefix + i);
+    }
+
+    logger.info("Check admin service {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkServiceCreated(adminServerPodName);
+
+    // check managed server services created
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Check managed server service {0} is created in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkServiceCreated(managedServerPrefix + i);
+    }
+  }
+
+  @Test
+  @Order(2)
+  @DisplayName("Create an ingress per domain")
+  public void testCreatingIngress() {
+    // helm install parameters for ingress
+    ingressParam = new HelmParams()
+            .releaseName(domainUID + "-ingress")
+            .namespace(domainNamespace)
+            .chartDir(projectRoot + "/../kubernetes/samples/charts/ingress-per-domain");
+
+    assertThat(createIngress(ingressParam, domainUID, domainUID + ".org"))
+            .as("Test createIngress returns true")
+            .withFailMessage("createIngress() did not return true")
+            .isTrue();
+
+    // check the ingress is created
+    String ingressName = domainUID + "-traefik";
+    String cmd = String.format("kubectl get ingress %1s -n %2s", ingressName, domainNamespace);
+    assertThat(getExpectedResult(cmd, ingressName))
```

Comment:
instead of `getExpectedResult` which I assume runs a kubectl command - use the `getIngress` action if we have one, if not add it and use it

Reply:
added a new method using Kubernetes API calls directly.